### ISSUE TITLE
"autosubmit create" command killed before it finishes

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#148]
In GitLab by @fmassonn on Jun 21, 2016, 12:32

Hi,

I have a somewhat annoying problem. My experiment `a06q` has a lot of jobs:
1 start date x 360 chunks x 24 members x 5 jobs per chunk  = more than 50 000 jobs 

I'm trying to create the experiment but it is killed because it takes too long. Then I'm opening an interactive session but get this message:

```
fmassonn@moore: ~ >> autosubmit create a06q
[CRITICAL] The directory /home/Earth/fmassonn/debug/autosubmit/a06q is needed and does not exist.
[WARNING] Does an experiment with the given id exist?
```

Any suggestion?

François

**Comment by kinow** [comment]
In GitLab by @ jlope2 on Jun 21, 2016, 15:07

Which error are you having?

Did you try to create this experiment with no-plot option? 

Maybe it's a memory problem..

**Comment by kinow** [comment]
In GitLab by @fmassonn on Jun 21, 2016, 15:51

This is what I tried:

```
fmassonn@ bscearth321: ~ >> autosubmit create a06q -np
Preparing .lock file to avoid multiple instances with same expid.

Checking configuration files...
autosubmit_a06q.conf OK
platforms_a06q.conf OK
jobs_a06q.conf OK
expdef_a06q.conf  OK
Configuration files OK

Using project folder: /esnas/autosubmit/a06q/proj
Loading parameters...

Creating joblist...
Creating jobs...
Adding dependencies...
Removing redundant dependencies...

Killed
```

**Comment by kinow** [comment]
In GitLab by @ jlope2 on Jun 21, 2016, 17:44

It seems that you are having a memory problem, so.. by the moment, there's nothing to do. Luckily since last days we are working on memory issues, so we're working in order to reduce the memory consumption of Autosubmit.

However, we will ask you for testing a development branch with an optimised version of Autosubmit, probably soon.

**Comment by kinow** [comment]
In GitLab by @dmanubens on Jun 23, 2016, 18:35

Dear @fmassonn 
We have development branch ready for testing. 
in order to test it you will need a virtualenv.
```
module unload autosubmit
mkdir ~/venvs/as_361
virtualenv ~/venvs/as_361
source ~/venvs/as_361/bin/activate
```
then you have to install the development branch
```
pip install https://earth.bsc.es/gitlab/es/autosubmit/repository/archive.zip?ref=develop-pkl-2-db
autosubmit -v
```
should tell you 3.6.1
Then before creating make a backup of your pkl and try creating with -np
```
cp job_list_a06q.pkl ~/job_list_a06q.bkp
autosubmit create a06q -np
```
Tell us if it works :D

**Comment by kinow** [comment]
In GitLab by @fmassonn on Jun 23, 2016, 18:39

Thanks @dmanubens ,

I'm doing this for the next submissions I'm launching!

Cheers and thanks for the support

François

**Comment by kinow** [comment]
In GitLab by @fmassonn on Jun 24, 2016, 11:48

I have tried the procedure with experiment `a08a` that had the problem in one of the CLEAN jobs:

```
fmassonn@ bscearth321: ~/logs/autosubmit >> cat /home/Earth/fmassonn/logs/autosubmit/log_a08a_20160623190251_bscesautosubmit01 | grep seemed -B 2 -A 2
218 of 235 jobs remaining (04:02)
Job a08a_19580101_fc00_5_SIM is RUNNING
[WARNING] Job a08a_19580101_fc00_4_CLEAN seemed to be completed but there is no COMPLETED file
[WARNING] Job a08a_19580101_fc00_4_CLEAN in UNKNOWN status

--
218 of 235 jobs remaining (04:02)
kill: sending signal to 1963112 failed: No such process
[WARNING] Job a08a_19580101_fc00_4_CLEAN seemed to be completed but there is no COMPLETED file
Job a08a_19580101_fc00_4_CLEAN is FAILED
Job a08a_19580101_fc00_5_SIM is RUNNING
```

Your procedure worked perfectly except that I had to start with a `module purge` instead of `module unload autosubmit`, simply because there were still other modules (like Python) messing up with the virtual environment.

I created the experiment successfully:
```
(as_361)fmassonn@ bscearth321: ~/logs/autosubmit >> autosubmit create a08a
Preparing .lock file to avoid multiple instances with same expid.

Checking configuration files...
autosubmit_a08a.conf OK
platforms_a08a.conf OK
jobs_a08a.conf OK
expdef_a08a.conf  OK
Configuration files OK

Using project folder: /esnas/autosubmit/a08a/proj
Loading parameters...

Creating joblist...
Creating jobs...
Adding dependencies...
Removing redundant dependencies...

Saving joblist...

Ploting joblist...
Plotting...
Plot created at /esnas/autosubmit/a08a/plot/a08a_20160624_1136.pdf
START /usr/bin/evince /esnas/autosubmit/a08a/plot/a08a_20160624_1136.pdf
Unable to init server: Could not connect: Connection refused
Cannot parse arguments: Cannot open display:

Job list created succesfully
Remember to MODIFY the MODEL config files!

```


but unfortunately was not able to recover what had already been done:

```
(as_361)fmassonn@ bscearth321: ~/logs/autosubmit >> expid=a08a; nc^Cup autosubmit run ${expid} >& ~/logs/autosubmit/log_${expid}_as-361_`date +%Y%m%d%H%M%S`_${HOST} &
(as_361)fmassonn@ bscearth321: ~/logs/autosubmit >> autosubmit recovery a08a
Recovering experiment a08a
[CRITICAL] Unhandled exception on Autosubmit: AutosubmitConfig instance has no attribute '_conf_parser'
Traceback (most recent call last):
  File "/home/Earth/fmassonn/venvs/as_361/lib/python2.7/site-packages/autosubmit/autosubmit.py", line 302, in parse_args
    return Autosubmit.recovery(args.expid, args.save, args.all, args.hide)
  File "/home/Earth/fmassonn/venvs/as_361/lib/python2.7/site-packages/autosubmit/autosubmit.py", line 949, in recovery
    job_list = Autosubmit.load_job_list(expid, as_conf)
  File "/home/Earth/fmassonn/venvs/as_361/lib/python2.7/site-packages/autosubmit/autosubmit.py", line 2001, in load_job_list
    Autosubmit._get_job_list_persistence(expid, as_conf))
  File "/home/Earth/fmassonn/venvs/as_361/lib/python2.7/site-packages/autosubmit/autosubmit.py", line 1883, in _get_job_list_persistence
    storage_type = as_conf.get_storage_type()
  File "/home/Earth/fmassonn/venvs/as_361/lib/python2.7/site-packages/autosubmit/config/config_common.py", line 790, in get_storage_type
    return self.get_option(self._conf_parser, 'storage', 'TYPE', 'db').lower()
AttributeError: AutosubmitConfig instance has no attribute '_conf_parser'

```

Hence I cannot check that this branch of autosubmit solves the issue.

One question: why do I have to re-create the experiment in the virtual environment?

François

**Comment by kinow** [comment]
In GitLab by @dmanubens on Jun 27, 2016, 08:02

Hi @fmassonn could you add these two lines in the `autosubmit_a08a.conf`:
```
[storage]
TYPE = db
```
Recreating the experiment is a must since the idea is to get rid of the pkl and use a sqlite database instead (much more efficient in terms of memory footprint).

**Comment by kinow** [comment]
In GitLab by @dmanubens on Jun 27, 2016, 11:08

Hi @fmassonn sorry I have tested and actually the problem persists. @ jlope2 has committed a fix. You can update your virtual environment using: `pip install https://earth.bsc.es/gitlab/es/autosubmit/repository/archive.zip?ref=develop-pkl-2-db --upgrade --no-deps`

**Comment by kinow** [comment]
In GitLab by @fmassonn on Jun 27, 2016, 11:09

Hi,

I will try this at the next occasion but in the mean time I had relaunched my experiment that needs to progress. 

I'll use your fix at the next occasion!

Cheers,

François

**Comment by kinow** [comment]
In GitLab by @ jlope2 on Jul 8, 2016, 16:28

This issue was due to the high memory consumption of Autosubmit.

We've improved a lot the performance (in terms of memory consumption), so from the next release, the experiments like this will work well in both, workstations and autosubmit's machine.

Also experiments larger than this one (at least up to 200-300k jobs) will work well.

By the moment I close this issue, we will reopen it if other performance problems reappear.
